### PR TITLE
link web3 and verify web3

### LINKED ISSUE #1054: ts-tests for web3 identity verification
### Context
It seems we only have web2(twitter) verification ts-tests, we need web3 too (substrate/evm). F/E colleagues also look for some exemplary code.

---
:heavy_check_mark: Please set appropriate **labels** and **assignees** if applicable.

### PR DESCRIPTION
**Pull request by 0xverin**
add web3 types;
add link web3;
add verify web3

### CODE REVIEW

File: tee-worker/ts-tests/utils.ts [hunk 1]
```diff
@@ -23,6 +23,10 @@ import { EventRecord } from "@polkadot/types/interfaces";
 import { after, before, describe } from "mocha";
 import { randomAsHex } from "@polkadot/util-crypto";
 import { generateChallengeCode, getSinger } from "./web3/setup";
```
Comment (Kailai-Wang) [inline tee-worker/ts-tests/utils.ts]: I believe a typo in setup.ts, should be `getSigner`

File: tee-worker/ts-tests/identity.test.ts [hunk 2]
```diff
@@ -72,28 +131,133 @@ describeLitentry("Test Identity", (context) => {
...
+                context,
+                hexToU8a(challengeCode),
+                context.defaultSigner.addressRaw,
+                substrateIdentity
+            );
+
+            console.log("post verification msg to substrate: ", msg);
+            substrateValidationData!.Web3Validation!.Substrate!.message = msg;
+            signature_substrate = context.defaultSigner.sign(msg);
+            substrateValidationData!.Web3Validation!.Substrate!.signature!.Sr25519 =
```
Comment (jonalvarezz) [inline tee-worker/ts-tests/identity.test.ts:180]: qq: when would this be ED255 or EDCA?

File: tee-worker/ts-tests/utils.ts (hunk 1 above)
Comment (0xverin) [reply tee-worker/ts-tests/utils.ts]: Ok, I've changed it.

File: tee-worker/ts-tests/utils.ts [hunk 3]
```diff
@@ -74,9 +78,19 @@ export async function getTEEShieldingKey(
 
 export async function initIntegrationTestContext(
     workerEndpoint: string,
-    substrateEndpoint: string
+    substrateEndpoint: string,
+    ethereumEndpoint: string
 ): Promise<IntegrationTestContext> {
     const provider = new WsProvider(substrateEndpoint);
+    const ethersProvider = new ethers.providers.JsonRpcProvider(ethereumEndpoint);
```
Comment (ziming-zung) [inline tee-worker/ts-tests/utils.ts:85]: I think ethereumEndpoint is unnecessary

**Review by ziming-zung — APPROVED** [review]
Basically, it looks good to me.

File: tee-worker/ts-tests/identity.test.ts (hunk 2 above)
Comment (ziming-zung) [reply tee-worker/ts-tests/identity.test.ts:180]: @jonalvarezz I don't understand your question, can you explain it more clearly?:see_no_evil:

**Comment by Kailai-Wang** [comment]
I'll merge it first, we can have another PR in case we have something to change

File: tee-worker/ts-tests/utils.ts (hunk 3 above)
Comment (0xverin) [reply tee-worker/ts-tests/utils.ts:85]: Yeah, I'll delete that later.

**Pull request merged**